pkg/controllers/provisioning/suite_test.go

```diff
@@ -145,6 +147,26 @@ var _ = Describe("Provisioning", func() {
 				ExpectScheduled(ctx, env.Client, pod)
 			}
 		})
+		Context("Instance Type Selection", func() {
+			BeforeEach(func() {
+				ctx = injection.WithTestInstance(ctx, true)
+			})
+
+			AfterEach(func() {
+				ctx = injection.WithTestInstance(ctx, false)
+			})
+
+			It("should not select invalid instance types", func() {
+				pod := ExpectProvisioned(ctx, env.Client, selectionController, provisioningController, provisioner, test.UnschedulablePod(
+					test.PodOptions{
+						ResourceRequirements: v1.ResourceRequirements{Requests: v1.ResourceList{v1.ResourceCPU: resource.MustParse("3"), v1.ResourceMemory: resource.MustParse("3Gi")}},
+					},
+				))[0]
+				node := ExpectScheduled(ctx, env.Client, pod)
```

Comment:
I'd check the node.Labels[instancetype] for a positive scheduling test.

Reply:
By positive test you meant it will select the illy constructed instance type?